src/usage_metrics/resources/sqldatabase.py

```diff
@@ -46,6 +54,11 @@ def append_df_to_table(
         query = sa.select(*[tbl.c[c] for c in pk_cols])  # Only select PK cols
 
         with self.engine.begin() as conn:
+            # TODO: Right now this method does not work consistently for datetime
+            # columns, as the datetimes parsed by SQLite and Pandas are not identical in
+            # format. Currently, rerunning a partition will raise an error for tables
+            # with datetimes as primary keys.
```

Comment:
Is it possible to catch the error and provide this information in a new error?

Reply:
Can we convert the `existing_pks` to datetimes and than compare?

Reply:
This has to do with how SQLite and pandas handle datetimes differently, when the original data has no timestamp or a timestamp of "00:00:00.0000". It doesn't apply to any of the current data, as the PKs I was running into this problem with I converted into Dates to accurately reflect their actual dtype. So, this is a future problem I'd like to resolve when we do have an actual datetime index column in the data.